sugondat-chain/runtimes/sugondat-kusama/src/lib.rs

```diff
@@ -267,14 +274,244 @@ impl pallet_balances::Config for Runtime {
+
+impl<T: frame_system::Config> Convert<Multiplier, Multiplier> for BlobsFeeAdjustment<T>
+where
+    T: frame_system::Config,
+{
+    /// This function should be a pure function used to update NextFeeMultiplier
+    /// but will also has the side effect of update NextLenghtMultiplier
+    fn convert(previous_fee_multiplier: Multiplier) -> Multiplier {
+        // Update TargetBlockSize if needed
+        let all_extrinsic_len = <frame_system::Pallet<T>>::all_extrinsics_len();
+
+        if all_extrinsic_len > TargetBlockSize::get() {
+            // Increase the tracker if needed
+            let tracker = BlockSizeTracker::get();
+
+            // If the used_block_size is larger than the TargetBlockSize
+            // for more than IncreaseDeltaBlocks consecutive, then the TargetBlockSize
+            // will be increased by DeltaTargetBlockSize.
+            if tracker + 1 >= IncreaseDeltaBlocks::get() {
+                let current_target_block_size = TargetBlockSize::get();
+                TargetBlockSize::set(&(current_target_block_size + DeltaTargetBlockSize::get()));
+                BlockSizeTracker::set(&0);
+            }
+
+            BlockSizeTracker::set(&(tracker + 1));
+        } else {
+            // The logic for updating to a new TargetBlockSize currently requires IncreaseDeltaBlocks
+            // to be constantly larger than TargetBlockSize. However, it might be possible
+            // to implement a window where we reset the tracker only if the
+            // block size remains below the TargetBlockSize for a certain number of blocks
+            BlockSizeTracker::set(&0);
+        }
+
+        // Update NextLengthMultiplier
+
+        // To update the value will be used the same formula as TargetedFeeAdjustment,
+        // described here: https://research.web3.foundation/Polkadot/overview/token-economics#2-slow-adjusting-mechanism
+        //
+        // so this is mainly a copy paste of that function because it works on normalized mesurments,
+        // so if it is ref_time, proof_size or lenght of the extrinsic the mutliplier will be evaluated properly.
+        // The main problem is that TargetedFeeAdjustment::convert uses directly a call to the storage to extract
+        // the weight of the current block so there is no way to pass the lenght as input argument,
+        // here I will copy paste all the needed part to update properly NextLengthMultiplier
+
+        // Defensive only. The multiplier in storage should always be at most positive. Nonetheless
+        // we recover here in case of errors, because any value below this would be stale and can
+        // never change.
+
+        let previous_len_multiplier = NextLengthMultiplier::get();
+        let min_multiplier = MinimumMultiplierBlockSize::get();
+        let max_multiplier = MaximumMultiplierBlockSize::get();
+        // TODO: why?
+        let previous_len_multiplier = previous_len_multiplier.max(min_multiplier);
```

Comment:
I assume this is intended as a clamp in case the `NextLengthMultiplier` changes unexpectedly or the `MinimumMultiplierBlockSize` changes unexpectedly.

Reply:
Yes, it makes sense! In the end, this is just a way to ensure that the used multiplier will never be lower than `MinimumMultiplierBlockSize`! So `NextLengthMultiplier` is fetched from storage, but if there is such low traffic that it ends up being lower than expected, then these line recover it!